### PR TITLE
gitignore (new formula)

### PR DESCRIPTION
**Pull request by karan**
This commit adds the formula for a new command-line based app
called "joe". The namespace for "joe" is taken on homebrew,
hence "gitignore" is used here. Current version is 0.0.2.

### CODE REVIEW

**Comment by DomT4** [comment]
Heads up this [may not meet](https://github.com/Homebrew/homebrew/blob/master/share/doc/homebrew/Acceptable-Formulae.md#niche-or-self-submitted-stuff) the Acceptable Formulae guide. A 22 hour old tool with little coverage thus far likely falls under that.

That's not an insult, by any means, for clarification; the tool looks useful, and it appeals to my sense of laziness & being able to operate solely within the CLI, but I'm not sure this doesn't at this point count under self-submitted stuff as referenced in the AF doc.

Feel free to await maintainer review though, they may have a different viewpoint.

File: Library/Formula/gitignore.rb [hunk 1]
```diff
@@ -0,0 +1,33 @@
+class Gitignore < Formula
+  homepage "https://github.com/karan/joe"
+  url "https://github.com/karan/joe/archive/0.0.2.tar.gz"
+  version "0.0.2"
```
Comment (DomT4) [inline Library/Formula/gitignore.rb]: Version is probably redundant given the url.

**Comment by karan** [comment]
That sounds off. I've certainly seen some smaller projects on brew, which makes me happy because of the easy of installation it adds.

Anyways, I'll wait for what maintainers say.

**Comment by bfontaine** [comment]
Note that with Homebrew you can maintain your own repo (“tap”), where you can add any formula you want. See for example [bfontaine/homebrew-utils](https://github.com/bfontaine/homebrew-utils#homebrew-utils), a repo I use for my personal utilities which wouldn’t be accepted in the main repo because Homebrew “frown[s] on authors submitting their own work unless it is very popular”.
Users just need to run `brew tap <your repo>` and they’ll be able to install your formulae.

File: Library/Formula/gitignore.rb [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+    end
+
+    system "python", *Language::Python.setup_install_args(libexec)
+
+    bin.install Dir["#{libexec}/bin/*"]
+    bin.env_script_all_files(libexec/"bin", :PYTHONPATH => ENV["PYTHONPATH"])
+  end
+
+  test do
+    system "#{bin}/joe", "--version"
```
Comment (mikemcquaid) [inline Library/Formula/gitignore.rb]: Can this test be modified to do something more substantial than e.g. `--version` or `--help`? See `cmake.rb` for an example of an application formula with a good test and `tinyxml2.rb` for an example of a library formula with a good test. Thanks!

**Comment by mikemcquaid** [comment]
I think this warrants passing on until it's a little newer (even like a week or so). Happy to leave it open until then, though.

**Comment by DomT4** [comment]
> That sounds off.

No malice or insult intended, by me or the document. I think the document is merely trying to make the point that it's difficult enough finding people to maintain 2.6k formulae as it is :wink:.

**Comment by jacknagel** [comment]
I would suggest choosing something other than "gitignore" as the package name.

**Comment by karan** [comment]
@jacknagel Just curious, why do you say so? (newbie to writing formulae here). :smile:

**Comment by karan** [comment]
Well this is not what I wanted. Closing this to open a new PR!

**Pull request closed without merging**